src/Tasks.UnitTests/RoslynCodeTaskFactory_Tests.cs

```diff
@@ -1068,6 +1068,61 @@ public override bool Execute()
             }
         }
 
+        [Fact]
+        public void RoslynCodeTaskFactory_UsingAPI()
+        {
+            string text = $@"
+<Project>
+
+  <UsingTask
+    TaskName=""Custom1""
+    TaskFactory=""RoslynCodeTaskFactory""
+    AssemblyFile=""$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll"" >
+    <ParameterGroup>
+      <SayHi ParameterType=""System.String"" Required=""true"" />
+    </ParameterGroup>
+    <Task>
+      <Reference Include=""{typeof(Enumerable).Assembly.Location}"" />
+      <Code Type=""Fragment"" Language=""cs"">
+        <![CDATA[
+        string sayHi = ""Hello "" + SayHi;
+        Log.LogMessage(sayHi);
+        ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+
+    <Target Name=""Build"">
+        <Custom1 SayHi=""World"" />
+    </Target>
+
+</Project>";
+
+            using var env = TestEnvironment.Create();
+#if !FEATURE_RUN_EXE_IN_TESTS
+            RunnerUtilities.ApplyDotnetHostPathEnvironmentVariable(env);
+#endif
+
+            var project = env.CreateTestProjectWithFiles("p1.proj", text);
+
+            var logger = project.BuildProjectExpectSuccess();
+#if !FEATURE_RUN_EXE_IN_TESTS
+            var filter = "dotnet path is ";
+#else
+            var filter = "Compiling task source code";
+
+#endif
```

Comment:
I don't think I understand the `#if` here. Can you explain why the behavior is conditional? Should we make the test netcore-only?

Reply:
Agree. The fix is for .NET version of msbuild. It's not necessary to test .net framework.